### DISCUSSION TITLE
How can I know if the current execution is a retry?

### DISCUSSION THREAD

**Discussion by sryza** [dagster-io/dagster#12583] [Q&A] [answered]
For an Op retry policy, how can we know inside the op context if the current execution is a retry?

**Comment by sryza** [comment, marked as answer]
You can use [context.retry_number](https://docs.dagster.io/_apidocs/execution#dagster.OpExecutionContext.retry_number).

```python
@op(retry_policy=RetryPolicy(max_retries=2))
def my_op(context):
    is_retry = context.retry_number > 0
    ...
```

**Reply by RWiggers** [reply]
This isn’t working for me. Every time an op fails, Dagster seems to create a new context and resets `retrying_number` back to zero
